src/react/hooks/__tests__/useLoadableQuery.test.tsx

```diff
@@ -0,0 +1,4848 @@
+
+  // initial render
+  await Profiler.takeRender();
+  // load query
+  await Profiler.takeRender();
+
+  {
+    const { snapshot } = await Profiler.takeRender();
+
+    expect(snapshot.result).toEqual({
+      data: { greeting: "Hello" },
+      error: undefined,
+      networkStatus: NetworkStatus.ready,
+    });
+  }
+
+  await act(() => user.click(screen.getByText("Change error policy")));
+  await act(() => user.click(screen.getByText("Refetch greeting")));
+
+  // change error policy
+  await Profiler.takeRender();
+  // refetch
+  await Profiler.takeRender();
+
+  {
+    const { snapshot, renderedComponents } = await Profiler.takeRender();
+
+    // Ensure we aren't rendering the error boundary and instead rendering the
+    // error message in the hook component.
+    expect(renderedComponents).not.toContain(ErrorFallback);
+    expect(snapshot.result).toEqual({
+      data: { greeting: "Hello" },
+      error: new ApolloError({ graphQLErrors: [new GraphQLError("oops")] }),
+      networkStatus: NetworkStatus.error,
+    });
+  }
+});
+
+it("applies `context` on next fetch when it changes between renders", async () => {
+  interface Data {
+    phase: string;
+  }
+
+  const query: TypedDocumentNode<Data, never> = gql`
+    query {
+      phase
+    }
+  `;
+
+  const link = new ApolloLink((operation) => {
+    return Observable.of({
+      data: {
+        phase: operation.getContext().phase,
+      },
+    });
+  });
+
+  const client = new ApolloClient({
+    link,
+    cache: new InMemoryCache(),
+  });
+
+  const Profiler = createDefaultProfiler<Data>();
+  const { SuspenseFallback, ReadQueryHook } =
+    createDefaultProfiledComponents(Profiler);
+
+  function App() {
+    const [phase, setPhase] = React.useState("initial");
+    const [loadQuery, queryRef, { refetch }] = useLoadableQuery(query, {
+      context: { phase },
+    });
+
+    return (
+      <>
+        <button onClick={() => setPhase("rerender")}>Update context</button>
+        <button onClick={() => refetch()}>Refetch</button>
+        <button onClick={() => loadQuery()}>Load query</button>
+        <Suspense fallback={<SuspenseFallback />}>
+          {queryRef && <ReadQueryHook queryRef={queryRef} />}
+        </Suspense>
+      </>
+    );
+  }
+
+  const { user } = renderWithClient(<App />, {
+    client,
+    wrapper: ({ children }) => <Profiler>{children}</Profiler>,
+  });
+
+  await act(() => user.click(screen.getByText("Load query")));
+
+  // initial render
+  await Profiler.takeRender();
+  // load query
+  await Profiler.takeRender();
+
+  {
+    const { snapshot } = await Profiler.takeRender();
+
+    expect(snapshot.result!.data).toEqual({
+      phase: "initial",
+    });
+  }
+
+  await act(() => user.click(screen.getByText("Update context")));
+  await act(() => user.click(screen.getByText("Refetch")));
+
+  // update context
+  await Profiler.takeRender();
+  // refetch
+  await Profiler.takeRender();
```

Comment:
Same as above (not repeating this suggestion after this)
```suggestion
  await act(() => user.click(screen.getByText("Update context")));
  await Profiler.takeRender();

  await act(() => user.click(screen.getByText("Refetch")));
  await Profiler.takeRender();
```

Reply:
Done! [935ff7c](https://github.com/apollographql/apollo-client/pull/11300/commits/935ff7cf2e5d2ba03285628519f67c1073b659de)